### DISCUSSION TITLE
OIDCWiremockTestResource and application.yml

### DISCUSSION THREAD

**Discussion by Tusenka** [quarkusio/quarkus#30079] [Community]
OIDCWiremockTestResource and application.yml 
Could you please get OIDCWiremockTestResource

What is the best way to integrate OIDCWiremockTestResource with password and user authentification?
There are five pathways in quarks OIDC WireMock:
CLIENT
PASSWORD
CODE/EXCHANGE/REFRESH

I am using CODE the OIDCWIremockTestResource works fine.
When I try to use password or secret it throw out 401 exception from the rest client. On token retrivement it wrouts that endpoint ${keycloak.url}/realms/quarkus doesn't match as it expects authentification_code

May I need some additional setup provided for the server? I just Autowire it and use. 

My config:
```
quarkus.oidc.auth-server-url=${keycloak.url}/realms/quarkus/
quarkus.oidc.client-id=quarkus-app
quarkus.oidc.credentials.secret=secret

quarkus.oidc-client.auth-server-url=${quarkus.oidc.auth-server-url}
quarkus.oidc-client.client-id=${quarkus.oidc.client-id}
quarkus.oidc-client.credentials.secret=${quarkus.oidc.credentials.secret}
quarkus.oidc-client.grant.type=password
quarkus.oidc-client.grant-options.password.username=alice
quarkus.oidc-client.grant-options.password.password=alice

quarkus.oidc-client.early-tokens-acquisition=false

io.quarkus.oidc.client.ProtectedResourceService/mp-rest/url=http://localhost:8081/protected

quarks.oidc-client.early-tokens-acquisition=false

io.quarkus.oidc.client.ProtectedResourceService/mp-rest/url=http://localhost:8081/protected
````

My compact code 
`  oidc-client:
    credentials:
      secret: secret
    grant:
      type: code
    application-type: service
    auth-server-url: ${keycloak.url}/realms/quarkus
    id: quarkus-app
    client-id: quarkus-app`

Maybe I need to do some additional adjustments to the OIDCWIremockTestResource?
I am using WireMock for the application to mock requests from the client.
 
Thank you, Irina

**Comment by Felixwanbar** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
